### DISCUSSION TITLE
How to stop realtime websocket started thru supabase.js .subscribe

### DISCUSSION THREAD

**Discussion by GaryAustin1** [supabase/supabase#4447] [Questions] [answered]
I've decided a better approach to handle realtime is to detect an idle browser tab (hidden >5 minutes) and then shut down realtime including the socket with a flag to reload potential data changes and restart realtime if browser tab goes visible again.

Closing all subscriptions does not appear to actually shut down the open socket (which keeps using bandwidth and depending on browser starts getting errors).

Anyone know a call to terminate realtime and the socket (not just the subscriptions)?

**Comment by w3b6x9** [comment, marked as answer]
@GaryAustin1 RealtimeClient has a [disconnect](https://github.com/supabase/realtime-js/blob/7983bb601f7d7804438faa30b6f52a2596f4e8d1/src/RealtimeClient.ts#L141) that you can use to terminate the socket. Give that a try.

**Reply by GaryAustin1** [reply]
I've opened a bug report in supabase/supabase on supabase.js issue with not closing out connection as I think there is a bug in that code...
But the only way I can figure out to call realtime disconnect is something like this:

subscriptions = supabase.getSubscriptions()
if (subscriptions.length) subscriptions[0].socket.disconnect()

As the realtime disconnect works on sockets, but supabase.js deals with connections/subpscriptions....
